### PR TITLE
Update repr docstring to hint what it stands for.

### PR DESCRIPTION
**Pull request by nathanrboyer**
Changed first line of `repr` docstring from "Create a string from any value ..." to "Create a string representation of any value ...".

This function is referenced in the [manual on output functions](https://docs.julialang.org/en/v1.12-dev/manual/types/#Output-function-summary), but I could not find what it stands for or means. I thought it was a typo of "repl" until "representation" finally came to me. I'm assuming that is the intent of the function name?

### CODE REVIEW

File: base/strings/io.jl [hunk 1]
```diff
@@ -250,7 +250,7 @@ print(io::IO, s::Union{String,SubString{String}}) = (write(io, s); nothing)
 """
     repr(x; context=nothing)
 
-Create a string from any value using the 2-argument `show(io, x)` function.
+Create a string representation of any value using the 2-argument `show(io, x)` function.
```
Comment (IanButterworth) [inline base/strings/io.jl:253]: Is it correct to go further and mention the idea of `eval(Base.Meta.parse(repr(x))) == x`?
I'm not sure whether it's strictly enforced though?
Something like this?
```suggestion
Create a string representation of any object using the 2-argument `show(io, x)` function.
In most cases `eval(Base.Meta.parse(repr(x))) == x` will hold true.
```

Comment (LilithHafner) [reply base/strings/io.jl:253]: The language used in the `show` docstring is "should be parseable Julia code when possible".  I like your suggestion to add that property.

I don't see why we should prefer "object" over "value".

**Review by LilithHafner — APPROVED** [review]

File: base/strings/io.jl (hunk 1 above)
Comment (IanButterworth) [reply base/strings/io.jl:253]: I thought object was a superset of value but perhaps that's not right. I'll update my suggestion above

File: base/strings/io.jl [hunk 2]
```diff
@@ -250,7 +250,8 @@ print(io::IO, s::Union{String,SubString{String}}) = (write(io, s); nothing)
 """
     repr(x; context=nothing)
 
-Create a string from any value using the 2-argument `show(io, x)` function.
+Create a string representation of any value using the 2-argument `show(io, x)` function, which aims to produce a string that is parseable Julia code where possible. i.e.
+`eval(Meta.parse(repr(x))) == x` should hold true.
```
Comment (IanButterworth) [inline base/strings/io.jl:254]: Formatting
```suggestion
Create a string representation of any value using the 2-argument `show(io, x)` function, 
which aims to produce a string that is parseable Julia code, where possible. 
i.e. `eval(Meta.parse(repr(x))) == x` should hold true.
```

File: base/strings/io.jl [hunk 3]
```diff
@@ -182,7 +182,7 @@ highly efficient, then it may make sense to add a method to `string` and
 define `print(io::IO, x::MyType) = print(io, string(x))` to ensure the
 functions are consistent.
 
-See also: [`String`](@ref), [`repr`](@ref), [`sprint`](@ref), [`show`](@ref @show).
```
Comment (LilithHafner) [inline base/strings/io.jl:185]: Is this a typo?

Comment (nathanrboyer) [reply base/strings/io.jl]: Oh whoops!

**Comment by LilithHafner** [comment]
Thanks, @nathanrboyer! This should help folks work with display and printing more easily.

**Pull request merged**